core/commands/config.go

```diff
@@ -524,33 +518,57 @@ func replaceConfig(r repo.Repo, file io.Reader) error {
 		return errors.New("private key in config was not a string")
 	}
 
-	cfg.Identity.PrivKey = pkstr
+	newCfg.Identity.PrivKey = pkstr
+
+	// Handle Pinning.RemoteServices (API.Key of each service is a secret)
 
-	// Handle Pinning.RemoteServices (ApiKey of each service is secret)
-	// Note: these settings are opt-in and may be missing
+	newServices := newCfg.Pinning.RemoteServices
+	oldServices, err := getRemotePinningServices(r)
+	if err != nil {
+		return tryRemoteServiceApiErr
+	}
 
-	if len(cfg.Pinning.RemoteServices) != 0 {
+	// fail fast if service lists are obviously different
+	if len(newServices) != len(oldServices) {
 		return tryRemoteServiceApiErr
 	}
 
-	// detect if existing config has any remote services defined..
-	if remoteServicesTag, err := getConfig(r, config.RemoteServicesSelector); err == nil {
+	// re-apply API details and confirm every modified service already existed
+	for name, oldSvc := range oldServices {
+		if newSvc, hadSvc := newServices[name]; hadSvc {
+			// fail if input changes any of API details
+			// (interop with config show: allow Endpoint as long it did not change)
+			if len(newSvc.API.Key) != 0 || (len(newSvc.API.Endpoint) != 0 && newSvc.API.Endpoint != oldSvc.API.Endpoint) {
```

Comment:
WDYT about adding more specific errors and/or docs here so people know what they need to remove from their config files before trying to do an `ipfs config replace`?

Reply:
Tried to go that route, but there are too many variants of things that can go wrong with config of services defined in  `Pinning.*`  that its easier to return `tryRemoteServiceApiErr` for all of them. 

This indicates that the problem is related to pinning service config, suggesting to use of `pin remote service` instead  without going into too much detail hy or what went wrong – I believe its the most useful error for end user.

Reply:
I added some useful error text that is more informative.